pkg/operator/encryption/key_controller_encryption_test.go

```diff
@@ -65,6 +65,123 @@ func TestEncryptionKeyController(t *testing.T) {
+					break
+				}
+				wasSecretValidated := false
+				for _, action := range actions {
+					if action.Matches("create", "secrets") {
+						createAction := action.(clientgotesting.CreateAction)
+						actualSecret := createAction.GetObject().(*corev1.Secret)
+						expectedSecret := createEncryptionKeySecretWithKeyFromExistingSecret(targetNamespace, targetGR, 6, actualSecret)
+						if !equality.Semantic.DeepEqual(actualSecret, expectedSecret) {
+							ts.Errorf(diff.ObjectDiff(actualSecret, expectedSecret))
+						}
+						if err := validateEncryptionKey(actualSecret); err != nil {
+							ts.Error(err)
+						}
+						wasSecretValidated = true
+						break
+					}
+				}
+				if !wasSecretValidated {
+					ts.Errorf("the secret wasn't created and validated")
+				}
+			},
+		},
+
+		// scenario 5: checks if a new write key is not created given that the previous one was migrated and the new write key already exists.
+		{
+			name: "no-op when the previous key was migrated and the current one is valid but hasn't been observed (no read/write annotations)",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			initialSecrets: []runtime.Object{
+				createExpiredMigratedEncryptionKeySecretWithRawKey("kms", schema.GroupResource{"", "secrets"}, 5, []byte("61def964fb967f5d7c44a2af8dab6865")),
+				createEncryptionKeySecretWithRawKey("kms", schema.GroupResource{"", "secrets"}, 6, []byte("61def964fb967f5d7c44a2af8dab6865")),
+			},
+			targetNamespace: "kms",
+			expectedActions: []string{"list:secrets:openshift-config-managed"},
+		},
+
+		// scenario 6: checks if a new secret write key with ID equal to "101" is created because the previous (with ID equal to "100") one was migrated.
+		//             note that IDs of keys (not secrets) cannot exceed 100
+		{
+			name: "creates a new write key because the previous one expired - overflow",
+			targetGRs: map[schema.GroupResource]bool{
+				schema.GroupResource{Group: "", Resource: "secrets"}: true,
+			},
+			initialSecrets: []runtime.Object{
+				createExpiredMigratedEncryptionKeySecretWithRawKey("kms", schema.GroupResource{"", "secrets"}, 100, []byte("61def964fb967f5d7c44a2af8dab6865")),
+			},
+			targetNamespace: "kms",
+			expectedActions: []string{"list:secrets:openshift-config-managed", "create:secrets:openshift-config-managed"},
+			validateFunc: func(ts *testing.T, actions []clientgotesting.Action, targetNamespace string, targetGRs map[schema.GroupResource]bool) {
+				var targetGR schema.GroupResource
+				for targetGR = range targetGRs {
+					break
+				}
+				wasSecretValidated := false
+				for _, action := range actions {
+					if action.Matches("create", "secrets") {
+						createAction := action.(clientgotesting.CreateAction)
+						actualSecret := createAction.GetObject().(*corev1.Secret)
+						expectedSecret := createEncryptionKeySecretWithKeyFromExistingSecret(targetNamespace, targetGR, 101, actualSecret)
```

Comment:
@enj we only limit the length for [the keys](https://github.com/enj/cluster-kube-apiserver-operator/pull/9/files#diff-191c287b0df900c918ce24e1d4928559R104) that end up in an EncryptionConfig, the names of the actual secrets are not limited. I think that [the intention](https://github.com/enj/cluster-kube-apiserver-operator/blob/enj/f/enc_config/pkg/operator/encryption/shared_encryption.go#L167) was to limit every resource, is this something we should improve?

Reply:
we decided to remove `keyID%100`, i'm going to follow up this change.